doc/user-guide.md

```diff
@@ -509,9 +510,11 @@ To configure secure HTTPS connections for your Knative deployment, see [Configur
 
 ### Kubernetes Application Navigator (kAppNav) support
 
-By default, Runtime Component Operator configures the Kubernetes resources it generates to allow automatic creation of an application definition by [kAppNav](https://kappnav.io/), Kubernetes Application Navigator. You can easily view and manage the deployed resources that comprise your application using Application Navigator. You can disable auto-creation by setting `createAppDefinition` to `false`.
+By default, Runtime Component Operator configures the Kubernetes resources it generates to allow automatic creation of an [Application definition](https://github.com/kubernetes-sigs/application) with the `applicationName` parameter as the `Application` CR name. The automatic creation is done by the [kAppNav](https://kappnav.io/), Kubernetes Application Navigator. You can easily view and manage the deployed resources that comprise your application using Application Navigator. You can disable auto-creation by setting `createAppDefinition` to `false`. 
 
-To join an existing application definition, disable auto-creation and set the label(s) needed to join the application on `RuntimeComponent` CR. See [Labels](#labels) section for more information.
+To join an existing application definition in the `RuntimeComponent` CR namespace, ensure the `applicationName` parameter is set to the name of the `Application` CR you want to join to. To join an existing application definition in another namespace, ensure that the `createAppDefinition` parameter is set to `false` and the `applicationName` parameter is set to the name of the existing `Application` CR you want to join to.
+
+The operator tries to find an `Application` CR with the `applicationName` parameter in the `RuntimeComponent` CR namespace. If it fails to find any, then it lists all the `Application` CRs with the value of the `applicationName` parameter as the name and the `RuntimeComponent` CR namespace listed in the value of the `kappnav.component.namespaces` annotation. If it finds any `Application` CR with the specified criteria, it adds labels listed in the `spec.selector.matchLabels` parameter to the `RuntimeComponent` CR. When all these steps fails, if the `createAppDefinition` is not set to `false`, the operator configures the Kubernetes resources it generates to allow automatic creation of an `Application` definition.
```

Comment:
This sentence is too long as it is 44 words: "If it fails to find any, then it lists all the `Application` CRs with the value of the `applicationName` parameter as the name and the `RuntimeComponent` CR namespace listed in the value of the `kappnav.component.namespaces` annotation." We want to break it up so that it is more readable for customers. How about something like: "If it fails to find any, then it lists all the `Application` CRs. The value of the `applicationName` parameter is the name. The `RuntimeComponent` CR namespace listed in the value of the `kappnav.component.namespaces` annotation is the namespace for the Application` CRs". Feel free to slack me so that we come up with the correct wording.

Change " When all these steps fails," to " When all these steps fail,".

Reply:
Discussed this through email and fixed in https://github.com/application-stacks/runtime-component-operator/pull/49/commits/b025566b79afd4799209156233426b295f655067.